app/ht_indexer/tests/ht_queue_service_tests/queue_consumer_test.py

```diff
@@ -21,141 +23,226 @@ def one_message():
-            to the dead letter queue and discarded from the main queue"""
+    def test_queue_consume_message(self, one_message, get_rabbit_mq_host_name):
+        """ Test for consuming a message from the queue
+        One message is published and consumed, then at the end of the test the queue is empty
+        """
+
+        producer_instance = QueueProducer(
+            user= "guest", password="guest", host=get_rabbit_mq_host_name,
+            queue_name="test_queue_consume_message", batch_size=1
+        )
 
-    # Clean up the queue
-    consumer_instance.conn.ht_channel.queue_purge(consumer_instance.queue_name)
+        consumer_instance = QueueConsumer(
+            "guest",
+            "guest",
+            get_rabbit_mq_host_name,
+            "test_queue_consume_message",
+            requeue_message=False,
+            batch_size=1
+        )
+
+        # Clean up the queue
+        consumer_instance.ht_channel.queue_purge(consumer_instance.queue_name)
 
-    for message in list_messages:
         # Publish the message
-        producer_instance.publish_messages(message)
+        producer_instance.publish_messages(one_message)
 
-    start_time = time.time()
-    for method_frame, _properties, body in consumer_instance.consume_message(inactivity_timeout=3):
+        for _method_frame, _properties, body in consumer_instance.consume_message(inactivity_timeout=5):
 
-        if method_frame:
-            try:
-                # Process the message
+            if _method_frame:
                 output_message = json.loads(body.decode('utf-8'))
 
-                # Use the message to raise an exception
-                if output_message.get("ht_id") == "5":
-                    # This will raise an exception
-                    logger.info(f"Message {output_message.get('ht_id')} processed successfully {1 / 0}")
-                # Acknowledge the message if the message is processed successfully
-                positive_acknowledge(consumer_instance.conn.ht_channel, method_frame.delivery_tag)
-            except Exception as e:
-                logger.info(
-                    f"Message {method_frame.delivery_tag} re-queued to {consumer_instance.queue_name}"
-                    f" with error: {e}")
-
-                # Reject the message
-                consumer_instance.reject_message(consumer_instance.conn.ht_channel, method_frame.delivery_tag)
-                current_time = time.time()
-
-                # This check was added to avoid the test to run indefinitely because the queue is not empty and
-                # it is stuck
-                if current_time - start_time > 60:
-                    logger.info("The test is taking too long: Test ended")
-                    break
-                time.sleep(1)
+                consumer_instance.positive_acknowledge(consumer_instance.ht_channel, _method_frame.delivery_tag)
+                time.sleep(3)  # Wait for the message to be acknowledged
```

Comment:
What happens if we don't have the sleep here? Does the same message get re-delivered multiple times? That seems odd if so, and I might want to look more carefully at exactly what's going on, the queue settings, etc.